core-bundle/src/Routing/ResponseContext/Factory/Provider/ResponseContextProviderInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\ResponseContext\Factory\Provider;
+
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContextInterface;
+
+interface ResponseContextProviderInterface
+{
+    /**
+     * @template T of ResponseContextInterface
+     * @psalm-param class-string<T> $responseContextClassName
+     * @return T
+     */
```

Comment:
This was probably added by accident for the `supports` function as well. But that one is returning a `bool` :slightly_smiling_face:

Reply:
Ah, right. I'm new to Psalm :D Is https://github.com/contao/contao/pull/2975/commits/44e2445cf658a638a8184109aa92a94cdcb55668 correct?

Reply:
LGTM!

Reply:
Ah, hold on. Psalm actually can't infer things on an interface like this. :unamused: 

see https://psalm.dev/r/020d52cea5 vs https://psalm.dev/r/af4b7450d7

IMO we would need to implement the 2nd version in this case (using `@template T` on a method level):
```php
interface ResponseContextProviderInterface
{
    /**
     * @template T of ResponseContextInterface
     * @psalm-param class-string<T> $responseContextClassName
     */
    public function supports(string $responseContextClassName): bool;

    /**
     * @template T of ResponseContextInterface
     * @psalm-param class-string<T> $responseContextClassName
     * @return T
     */
    public function create(string $responseContextClassName): ResponseContextInterface;
}
```